### PR TITLE
Add the possibility to write info to audit log

### LINKED ISSUE #3781: Email Notification: Some errors do not result in a Success-False in the audit log
If the email of the user notification can not be passed to the mail server, we get different errors:

https://github.com/privacyidea/privacyidea/blob/9528249a8de5b71afa11c82617e48c02bb8e74e5/privacyidea/lib/eventhandler/usernotification.py#L442

It looks like in some cases we do not get the correct return value.

# Analysis

Not all problems with sending emails can be fetched. However, if an error occurs during passing the email to the sending email server, the `ret` is processed correctly.

If the error occurs later, e.g. due to non existing mailboxes of the recipient, this problem is out of our control.

Nevertheless we can do the following.

# Implementation

enhance the `do` method of the event handlers to not only return a state of true or false but also return an additional info, which can be appended to `event_audit_data["info"]` here:

PRE:

https://github.com/privacyidea/privacyidea/blob/4fd7492ec86306d95511f70cfa8f0df9c8add367/privacyidea/lib/event.py#L91

POST:

https://github.com/privacyidea/privacyidea/blob/4fd7492ec86306d95511f70cfa8f0df9c8add367/privacyidea/lib/event.py#L134

### PR DESCRIPTION
**Pull request by cornelinux**
An event handler can now write additional info to the audit log.

Closes #3781

### CODE REVIEW

**Review by jona-samuel — CHANGES_REQUESTED** [review]
Is it possible to test this?
If there isn't a good way to test this, then it can get merged

**Comment by cornelinux** [comment]
I investigated the scenarios on a live example.

Since it also is not clear, what the smtplib returns in case of

a) a wrong server name (IP not exist)
b) a mailbox does not exist locally (after connect)

I will check to use `smtpmock`.

**Review by jona-samuel — CHANGES_REQUESTED** [review]
Looks good, but please fix the typo

**Comment by cornelinux** [comment]
@jona-samuel I fixed the typo and also added a further test for pre-event-handling.
The code cov should be fine, now.

File: tests/test_lib_eventhandler_usernotification.py [hunk 1]
```diff
@@ -119,6 +119,55 @@ def test_02_sendmail(self):
...
+
+        r = add_smtpserver(identifier="myserver", server="1.2.3.4", tls=False)
+        self.assertTrue(r > 0)
+
+        smtpmock.setdata(response={"user@localhost.localdomain": (450, "Mailbox not abailable")},
+                         support_tls=False)
+
+        g = FakeFlaskG()
+        audit_object = FakeAudit()
+        audit_object.audit_data["serial"] = "123456"
```
[the remark this answers is not in the document]
Comment (cornelinux) [reply tests/test_lib_eventhandler_usernotification.py:135]: Fixed in 5cf0c430a5cd37b64b37fb010aeeaa8858574a95

**Review by jona-samuel — APPROVED** [review]

**Pull request merged**